deploy/SkyInit.sol

```diff
@@ -54,4 +60,18 @@ library SkyInit {
         dss.chainlog.setAddress("SKY",     instance.sky);
         dss.chainlog.setAddress("MKR_SKY", instance.mkrSky);
     }
+
+    function initSupplySync(
+        DssInstance memory dss,
+        address supplySync
+    ) internal {
+        SkyLike sky = SkyLike(dss.chainlog.getAddress("SKY"));
+
+        require(SupplySyncLike(supplySync).mkr() == dss.chainlog.getAddress("MCD_GOV"), "SkyInit/mkr-does-not-match");
+        require(SupplySyncLike(supplySync).sky() == address(sky),                       "SkyInit/sky-does-not-match");
+        require(sky.allowance(supplySync, dss.chainlog.getAddress("MCD_PAUSE_PROXY")) == type(uint256).max, "SkyInit/allowance-not-set");
```

Comment:
What's the rationale for including this safety check? It doesn't hurt of course but it seems to go beyond the usual ctor param checks. Isn't it good enough to have this check done in the tests?